### PR TITLE
[IN-2468] Hardcode Xclogparser to 0.2.21 so we can install it on all relevant VMs

### PR DESCRIPTION
**Pull request by fehersanyi-bitrise**
### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [ ] The tool I added is stable, and __does not require frequent updates__. _Preinstalled tools on the LTS stacks
  are not updated, so if the tool requires frequent updates you should handle the installation on-demand (see: [Install Any Additional Tool - DevCenter](https://bitrise-io.github.io/devcenter/tips-and-tricks/install-additional-tools/) for more information)_
- [ ] I updated the corresponding tests if there were any.
- [ ] I added a version report line to [`system_report.sh`](/system_report.sh) for the new tool(s) I added
- [ ] I have added an entry to the CHANGELOG.md with a revisionID (the current date and the number of the change e.g.: 2019_10_11_1) and a brief description of the actual change
- [ ] I have updated the revisionID in [bitrise_profile](roles/profiles/files/bitrise_profile) **BITRISE_OSX_STACK_REV_ID** with the one I just created in the CHANGELOG.md

### CODE REVIEW

**Review by bitrise-tom — CHANGES_REQUESTED** [review]
Two questions.
1. How do you know that v 0.2.21 is installed?
2. what happens to `which` when `/usr/local/bin` isn't in the path (i know it __should__ be)

**Comment by fehersanyi-bitrise** [comment]
@bitrise-tom it is the first item added to PATH in the profiles
https://github.com/bitrise-io/osx-box-bootstrap/blob/104d08ac7f15dd36316a073c368906eb2eeaa415/roles/profiles/files/zshrc#L1

**Comment by fehersanyi-bitrise** [comment]
@bitrise-tom 
1. it shouldn't be installed at all when this role is run. the check is only there for idempotency 🤷‍♂️

**Comment by bitrise-tom** [comment]
this is the weekly 

> @bitrise-tom
> 
> 1. it shouldn't be installed at all when this role is run. the check is only there for idempotency man_shrugging

this is the weekly update cache role, so it's going to run every week

The idempotency isn't really working as if you update the version number for a new release, it won't even download if there is any file already there

**Comment by fehersanyi-bitrise** [comment]
@bitrise-tom then what do you recommend? 
maybe checking the version, and if there is none or it is not the version we want, then we install it? 🧐

File: roles/weekly-update-shared/tasks/main.yml [hunk 1]
```diff
@@ -67,11 +67,27 @@
...
-    url: 'https://github.com/thii/xcbeautify.git'
+- name: Check if Xclogparser is present
+  shell: which xclogparser
+  register: xclogparser_is_present
+  ignore_errors: true
+  no_log: true
+
+- name: get Xclogparser 0.2.21
+  get_url:
+    url: https://github.com/spotify/XCLogParser/releases/download/v0.2.21/XCLogParser-macOS-x86_64-v0.2.21.zip
```
Comment (peterheja-bitrise) [inline roles/weekly-update-shared/tasks/main.yml:78]:
```suggestion
    url: https://github.com/spotify/XCLogParser/releases/download/v0.2.21/XCLogParser-macOS-x86_64-v0.2.21.zip
    checksum: sha256:f74caec7c8710dcdb0f4089001fd62d14daa5d3922b9b69626d7b9ce7b8e33e9
```

**Review by peterheja-bitrise — CHANGES_REQUESTED** [review]
First of all: are we sure we want to hardcode `xclogparser` to a fixed version across all stacks? I'm not against it, but it might be troublesome to update it.
Although, in the long run this is definitely something I support!
(So, it is +1 from me, but we have to be careful around this - it might rot!)

Secondly, there are some issues:
Does not take into account if `xclogparser` is older, than what we want
We should check for the output to include the version we want.

Does not hardcode / check for checksum.
We should have on the get_url the checksum parameter
We can get it by
```shasum -a 256 XCLogParser-macOS-x86_64-v0.2.21.zip
f74caec7c8710dcdb0f4089001fd62d14daa5d3922b9b69626d7b9ce7b8e33e9  XCLogParser-macOS-x86_64-v0.2.21.zip```

File: roles/weekly-update-shared/tasks/main.yml [hunk 2]
```diff
@@ -67,11 +67,27 @@
   with_items:
     - "{{ gem_packages }}"
 
-# brew taps
-- name: brew tap xcbeautify
-  homebrew_tap:
-    name: thii/xcbeautify
-    url: 'https://github.com/thii/xcbeautify.git'
+- name: Check if Xclogparser is present
```
Comment (peterheja-bitrise) [inline roles/weekly-update-shared/tasks/main.yml:70]: This should register / check the version

File: roles/weekly-update-shared/tasks/main.yml [hunk 3]
```diff
@@ -67,11 +67,27 @@
...
+  shell: which xclogparser
+  register: xclogparser_is_present
+  ignore_errors: true
+  no_log: true
+
+- name: get Xclogparser 0.2.21
+  get_url:
+    url: https://github.com/spotify/XCLogParser/releases/download/v0.2.21/XCLogParser-macOS-x86_64-v0.2.21.zip
+    dest: /tmp/xclogpraser.zip
```
Comment (peterheja-bitrise) [inline roles/weekly-update-shared/tasks/main.yml:79]: `xclogpraser` not `xclogparser`

File: roles/weekly-update-shared/tasks/main.yml [hunk 4]
```diff
@@ -67,11 +67,27 @@
...
+  when: xclogparser_is_present.rc != 0
+
+- name: unarchive xclogpraser binary
```
Comment (peterheja-bitrise) [inline roles/weekly-update-shared/tasks/main.yml:82]: `xclogpraser` not `xclogparser`

File: roles/weekly-update-shared/tasks/main.yml [hunk 5]
```diff
@@ -67,11 +67,27 @@
...
+  unarchive:
+    src: /tmp/xclogpraser.zip
```
Comment (peterheja-bitrise) [inline roles/weekly-update-shared/tasks/main.yml:84]: `xclogpraser` not `xclogparser`

**Comment by fehersanyi-bitrise** [comment]
@peterheja-bitrise in my opinion this is the safest way to go at the moment, I would not introduce any questionable logic into the system, with extra maintenance cost. I don't think I need to give reasoning to why I think this.

**Review by peterheja-bitrise — APPROVED** [review]

File: roles/weekly-update-shared/tasks/main.yml [hunk 6]
```diff
@@ -67,11 +67,28 @@
   with_items:
     - "{{ gem_packages }}"
 
-# brew taps
-- name: brew tap xcbeautify
-  homebrew_tap:
-    name: thii/xcbeautify
-    url: 'https://github.com/thii/xcbeautify.git'
+- name: Check if Xclogparser is present
+  shell: xclogparser version | awk '{print $2}'
```
Comment (peterheja-bitrise) [inline roles/weekly-update-shared/tasks/main.yml:71]: Use full path

**Review by peterheja-bitrise — CHANGES_REQUESTED** [review]
Use full paths

File: roles/weekly-update-shared/tasks/main.yml [hunk 7]
```diff
@@ -67,11 +67,28 @@
...
+  get_url:
+    url: https://github.com/spotify/XCLogParser/releases/download/v0.2.21/XCLogParser-macOS-x86_64-v0.2.21.zip
+    checksum: sha256:f74caec7c8710dcdb0f4089001fd62d14daa5d3922b9b69626d7b9ce7b8e33e9
+    dest: /tmp/xclogparser.zip
+  when: xclogparser_is_present.rc != 0 or xclogparser_is_present != 0.2.21
+
+- name: unarchive xclogparser binary
+  unarchive:
+    src: /tmp/xclogparser.zip
+    dest: /usr/local/bin/xclogparser
```
Comment (bitrise-tom) [inline roles/weekly-update-shared/tasks/main.yml:86]: Is this correct?
I think this will unpack the binary to `/usr/local/bin/xclogparser/xclogparser`

Comment (fehersanyi-bitrise) [reply roles/weekly-update-shared/tasks/main.yml:86]: you are right

**Review by peterheja-bitrise — DISMISSED** [review]
LGTM

File: roles/weekly-update-shared/tasks/main.yml [hunk 8]
```diff
@@ -67,11 +67,28 @@
...
+  register: xclogparser_version
+  ignore_errors: true
+  no_log: true
+
+- name: get Xclogparser 0.2.21
+  get_url:
+    url: https://github.com/spotify/XCLogParser/releases/download/v0.2.21/XCLogParser-macOS-x86_64-v0.2.21.zip
+    checksum: sha256:f74caec7c8710dcdb0f4089001fd62d14daa5d3922b9b69626d7b9ce7b8e33e9
+    dest: /tmp/xclogparser.zip
+  when: xclogparser_is_present.rc != 0 or xclogparser_is_present != 0.2.21
```
Comment (bitrise-tom) [inline roles/weekly-update-shared/tasks/main.yml:81]: this conditional should be based on  `xclogparser_is_present.stdout`  to match the version number

File: roles/weekly-update-shared/tasks/main.yml [hunk 9]
```diff
@@ -67,11 +67,28 @@
...
+
+- name: unarchive xclogparser binary
+  unarchive:
+    src: /tmp/xclogparser.zip
+    dest: /usr/local/bin/
+    owner: vagrant
+    mode: "0644"
+    remote_src: true
+  become: true
+  when: xclogparser_is_present.rc != 0 or xclogparser_is_present != 0.2.21
```
Comment (bitrise-tom) [inline roles/weekly-update-shared/tasks/main.yml:91]: this conditional should be based on  `xclogparser_is_present.stdout`  to match the version number

**Review by bitrise-tom — CHANGES_REQUESTED** [review]

**Review by peterheja-bitrise — APPROVED** [review]
LGTM

File: roles/weekly-update-shared/tasks/main.yml [hunk 10]
```diff
@@ -67,11 +67,28 @@
...
+  register: xclogparser_version
+  ignore_errors: true
+  no_log: true
+
+- name: get Xclogparser 0.2.21
+  get_url:
+    url: https://github.com/spotify/XCLogParser/releases/download/v0.2.21/XCLogParser-macOS-x86_64-v0.2.21.zip
+    checksum: sha256:f74caec7c8710dcdb0f4089001fd62d14daa5d3922b9b69626d7b9ce7b8e33e9
+    dest: /tmp/xclogparser.zip
+  when: xclogparser_is_present.rc != 0 or xclogparser_is_present.stdout != 0.2.21
```
Comment (bitrise-tom) [inline roles/weekly-update-shared/tasks/main.yml:81]: I think you will need to quote the "0.2.21"

**Review by peterheja-bitrise — APPROVED** [review]
LGTM

**Review by bitrise-tom — APPROVED** [review]

**Pull request merged**